Cesium.CodeGen.Tests/UseCultureAttribute.cs

```diff
@@ -0,0 +1,89 @@
+public class UseCultureAttribute : BeforeAfterTestAttribute
+{
+    readonly Lazy<CultureInfo> culture;
+    readonly Lazy<CultureInfo> uiCulture;
+
+    CultureInfo originalCulture = null!;
+    CultureInfo originalUICulture = null!;
+
+    /// <summary>
+    /// Replaces the culture and UI culture of the current thread with
+    /// <paramref name="culture" />
+    /// </summary>
+    /// <param name="culture">The name of the culture.</param>
+    /// <remarks>
+    /// <para>
+    /// This constructor overload uses <paramref name="culture" /> for both
+    /// <see cref="Culture" /> and <see cref="UICulture" />.
+    /// </para>
+    /// </remarks>
+    public UseCultureAttribute(string culture)
+        : this(culture, culture) { }
+
+    /// <summary>
+    /// Replaces the culture and UI culture of the current thread with
+    /// <paramref name="culture" /> and <paramref name="uiCulture" />
+    /// </summary>
+    /// <param name="culture">The name of the culture.</param>
+    /// <param name="uiCulture">The name of the UI culture.</param>
+    public UseCultureAttribute(string culture, string uiCulture)
+    {
+        this.culture = new Lazy<CultureInfo>(() => new CultureInfo(culture, false));
+        this.uiCulture = new Lazy<CultureInfo>(() => new CultureInfo(uiCulture, false));
+    }
+
+    /// <summary>
+    /// Gets the culture.
+    /// </summary>
+    public CultureInfo Culture { get { return culture.Value; } }
+
+    /// <summary>
+    /// Gets the UI culture.
+    /// </summary>
+    public CultureInfo UICulture { get { return uiCulture.Value; } }
+
+    /// <summary>
+    /// Stores the current <see cref="Thread.CurrentPrincipal" />
+    /// <see cref="CultureInfo.CurrentCulture" /> and <see cref="CultureInfo.CurrentUICulture" />
+    /// and replaces them with the new cultures defined in the constructor.
+    /// </summary>
+    /// <param name="methodUnderTest">The method under test</param>
+    public override void Before(MethodInfo methodUnderTest)
+    {
+        originalCulture = Thread.CurrentThread.CurrentCulture;
+        originalUICulture = Thread.CurrentThread.CurrentUICulture;
+
+        Thread.CurrentThread.CurrentCulture = Culture;
+        Thread.CurrentThread.CurrentUICulture = UICulture;
+
+        CultureInfo.CurrentCulture.ClearCachedData();
+        CultureInfo.CurrentUICulture.ClearCachedData();
+    }
+
+    /// <summary>
+    /// Restores the original <see cref="CultureInfo.CurrentCulture" /> and
+    /// <see cref="CultureInfo.CurrentUICulture" /> to <see cref="Thread.CurrentPrincipal" />
+    /// </summary>
+    /// <param name="methodUnderTest">The method under test</param>
+    public override void After(MethodInfo methodUnderTest)
+    {
+        Thread.CurrentThread.CurrentCulture = originalCulture;
+        Thread.CurrentThread.CurrentUICulture = originalUICulture;
+
+        CultureInfo.CurrentCulture.ClearCachedData();
+        CultureInfo.CurrentUICulture.ClearCachedData();
+    }
+}
```

Comment:
This seems to be copy-paste from xUnit repository: https://github.com/xunit/samples.xunit/blob/5dc1d35a63c3394a8678ac466b882576a70f56f6/UseCulture/UseCultureAttribute.cs

Copying external code is a little compilcated (we have to follow the license), and it also does a bit more work that I want it to (allows using named cultures etc.).

I suggest rewriting this attribute, making it only work with `InvariantCulture`, and removal of the copy-paste altogether (so there's only your code in the file).